### DISCUSSION TITLE
How to make imageview clickable in toga

### DISCUSSION THREAD

**Discussion by wanglifeng95** [beeware/toga#1959] [Q&A]
Beeware is a great development tool. But I still encountered some confusion. The button can be clicked, but pictures cannot be added. And Imageview can display images, but how does it implement click events?

**Comment by freakboy3742** [comment]
There isn't currently any way to add an image to a button, or an `on_click` handler to a image.

This has been the subject of a lot of debate in the past; see the following links for the background and discussion:
* https://github.com/beeware/toga/issues/774
* https://github.com/beeware/toga/discussions/1343

The short version is that we're unlikely to allow an *Image* to be added to button, but we might allow for an *icon* to be added to a button. See the discussion for the (very significant) differences between the two.

**Reply by wanglifeng95** [reply]
Thank you very much for your reply and I have also reviewed the two links you recommended. Image click events and button addition icons are both considered important. 
On this website, I saw your head image and had an external impression of you as a smart and handsome person. Then I clicked on your image to further open and view your information. In computer, mobile app, or WEB, we often use images to have a visual understanding of people or things, and then click on the images for further actions. Image click events have many functions, I sincerely hope that you are willing to help achieve it.
Adding icons to buttons is just like the function of a toolbar. The text appears cold, and the icons make people feel comfortable and vivid. Therefore, toolbars need icons, and buttons also need icons. The difference is that buttons are more flexible than toolbars, and can be placed anywhere in the APP. 
I am very happy to see that you mentioned adding icons to the buttons. Thank you very much for your efforts. Will see icon buttons in the new Toga version in the second half of the year？

**Reply by wanglifeng95** [reply]
> There isn't currently any way to add an image to a button, or an `on_click` handler to a image.
> 
> This has been the subject of a lot of debate in the past; see the following links for the background and discussion:
> 
> * [How to place a button with an icon/image #774](https://github.com/beeware/toga/issues/774)
> * [[android] image button.on_press() possible? #1343](https://github.com/beeware/toga/discussions/1343)
> 
> The short version is that we're unlikely to allow an _Image_ to be added to button, but we might allow for an _icon_ to be added to a button. See the discussion for the (very significant) differences between the two.

In the link, you mentioned the "Tabs" OptionContainer. I have previously implemented a relatively simple one that is suitable for mobile and desktop platforms based on Toga's widgets. If you find it meaningful, I can post the code for you

**Reply by freakboy3742** [reply]
Thanks - but in Toga's core we're not interested in "faking" widgets that have full native implementations. Both Android and iOS provide a native "tab container"; our intention for Toga is to wrap these widgets, rather than trying to build substitutes from primitives.

**Reply by wanglifeng95** [reply]
> Thanks - but in Toga's core we're not interested in "faking" widgets that have full native implementations. Both Android and iOS provide a native "tab container"; our intention for Toga is to wrap these widgets, rather than trying to build substitutes from primitives.

Hope "tab container can be used on mobile devices as soon as possible, and see the icon buttons earlier. Thanks very much.

**Comment by wanglifeng95** [comment]
您好，我已收到来信，会尽快给予你回复。
